packages/eslint-plugin/src/rules/no-invalid-this.ts

```diff
@@ -0,0 +1,86 @@
+import { TSESTree } from '@typescript-eslint/experimental-utils';
+import baseRule from 'eslint/lib/rules/no-invalid-this';
+import {
+  InferOptionsTypeFromRule,
+  InferMessageIdsTypeFromRule,
+  createRule,
+  deepMerge,
+} from '../util';
+
+export type Options = InferOptionsTypeFromRule<typeof baseRule>;
+export type MessageIds = InferMessageIdsTypeFromRule<typeof baseRule>;
+
+const schema = deepMerge(
+  Array.isArray(baseRule.meta.schema)
+    ? baseRule.meta.schema[0]
+    : baseRule.meta.schema,
+  {
+    properties: {
+      capIsConstructor: {
+        type: 'boolean',
+        default: true,
+      },
+    },
+  },
+);
+
+export default createRule<Options, MessageIds>({
+  name: 'no-invalid-this',
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description:
+        'disallow `this` keywords outside of classes or class-like objects',
+      category: 'Best Practices',
+      recommended: false,
+      extendsBaseRule: true,
+    },
+    messages: baseRule.meta.messages,
+    schema: [schema],
```

Comment:
don't need to mutate the schema if you're not adding any options

```suggestion
export default createRule<Options, MessageIds>({
  name: 'no-invalid-this',
  meta: {
    type: 'suggestion',
    docs: {
      description:
        'disallow `this` keywords outside of classes or class-like objects',
      category: 'Best Practices',
      recommended: false,
      extendsBaseRule: true,
    },
    messages: baseRule.meta.messages,
    schema: baseRule.meta.schema,
```